File: backend/app/schemas/application.py
from app.models.models import ApplicationStage
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class ApplicationCreate(BaseModel):
    userid: str
    company_id: str | None = None
    role: str | None = None
    date_applied: datetime
    loc: str | None = None
    employment_type: str | None = None
    notes: str | None = None

class ApplicationUpdate(BaseModel):
    company_id: str | None = None
    stage: ApplicationStage | None = None
    loc: str | None = None
    employment_type: str | None = None
    notes: str | None = None

class ApplicationRead(BaseModel):
    id: int
    company_id: str | None = None
    role: str | None = None
    stage: ApplicationStage | None = None
    date_applied: datetime
    loc: str | None
    employment_type: str | None = None
    notes: str | None = None

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/types/application.ts
export type ApplicationStage = 
  | "applied"
  | "oa"
  | "interview"
  | "offer"
  | "rejected"
  | "withdrawn";

export interface Application {
  id: number;
  company_name: string | null;
  role: string | null;
  stage: ApplicationStage | null;
  date_applied: string;
  loc: string | null;
  employment_type: string | null;
  notes: string | null;
}